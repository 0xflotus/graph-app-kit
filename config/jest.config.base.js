module.exports = {
  rootDir: "../",
  collectCoverageFrom: ["src/**/*.{js,jsx}"],
  setupFiles: [
    "<rootDir>/config/polyfills.js",
    "<rootDir>/config/jest.polyfills.js"
  ],
  testEnvironment: "node",
  transform: {
    "^.+\\.(js|jsx)$": "<rootDir>/node_modules/babel-jest"
  },
  transformIgnorePatterns: ["[/\\\\]node_modules[/\\\\].+\\.(js|jsx)$"],
  moduleNameMapper: {
    "^react-native$": "react-native-web",
    "^graph-app-kit(.*)": "<rootDir>/src$1"
  },
  moduleFileExtensions: ["web.js", "js", "json", "web.jsx", "jsx", "node"]
};
