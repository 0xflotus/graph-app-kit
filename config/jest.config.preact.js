const base = require("./jest.config.base");

const out = Object.assign({}, base, {
  moduleNameMapper: Object.assign({}, base.moduleNameMapper, {
    "^react$": "preact-compat",
    "^react-dom$": "preact-compat"
  }),
  snapshotSerializers: ["preact-render-spy/snapshot"]
});

module.exports = out;
