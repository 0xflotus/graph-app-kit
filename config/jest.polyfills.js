global.document.body.createTextRange = jest.fn(() => ({
  getBoundingClientRect: jest.fn(() => ({})),
  getClientRects: jest.fn(() => [])
}));
