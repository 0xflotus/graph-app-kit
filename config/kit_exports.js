const path = require("path");
const fs = require("fs");

// Make sure any symlinks in the project folder are resolved:
// https://github.com/facebookincubator/create-react-app/issues/637
const appDirectory = fs.realpathSync(process.cwd());
const resolveApp = relativePath => path.resolve(appDirectory, relativePath);

module.exports = {
  "ui/AsciiTable": resolveApp("src/ui/AsciiTable"),
  "ui/Chart": resolveApp("src/ui/Chart"),
  "ui/Render": resolveApp("src/ui/Render"),
  "utils/Cypher": resolveApp("src/utils/Cypher"),
  "utils/DesktopIntegration": resolveApp("src/utils/DesktopIntegration"),
  "utils/DriverProvider": resolveApp("src/utils/DriverProvider"),
  "utils/GraphAppBase": resolveApp("src/utils/GraphAppBase")
};
