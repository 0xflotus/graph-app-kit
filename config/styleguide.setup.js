const sleep = (secs, shouldResolve = true, message) =>
  new Promise((resolve, reject) => {
    setTimeout(
      () => (shouldResolve ? resolve(message) : reject(new Error(message))),
      secs * 1000
    );
  });
global.sleep = sleep;

const resolvingDriver = (waitSecs = 0, resolveTo = "Resolved") =>
  driver(true, waitSecs, resolveTo);
global.resolvingDriver = resolvingDriver;

const rejectingDriver = (waitSecs = 0, rejectTo = "Rejected") =>
  driver(false, waitSecs, rejectTo);
global.rejectingDriver = rejectingDriver;

const driver = (shouldResolve = true, waitSecs = 0, message = "") => ({
  session: () => ({
    close: () => {},
    run: () => sleep(waitSecs, shouldResolve, message)
  })
});
global.driver = driver;

const neo4jDesktopApi = {
  getContext: () => Promise.resolve({ x: 1 }),
  onContextUpdate: fn => {
    fn(1);
  }
};
global.neo4jDesktopApi = neo4jDesktopApi;
