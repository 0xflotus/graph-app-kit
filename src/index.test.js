import React from "react";
import TestRenderer from "react-test-renderer";
import { utils, ui } from "./index";
import { mockDriver } from "../config/test_helpers";

it("loads Cypher", () => {
  const out = TestRenderer.create(
    <utils.Cypher query="x" driver={mockDriver()} render={() => null} />
  );
  expect(out.toJSON()).toEqual(null);
});
it("loads DriverProvider", () => {
  const out = TestRenderer.create(
    <utils.DriverProvider driver={mockDriver()}>
      <div />
    </utils.DriverProvider>
  );
  expect(out.toJSON()).toMatchSnapshot();
});
it("loads Render (with output)", () => {
  const out = TestRenderer.create(
    <ui.Render if={true}>
      <span>Hello</span>
    </ui.Render>
  );
  expect(out.toJSON()).toMatchSnapshot();
});
