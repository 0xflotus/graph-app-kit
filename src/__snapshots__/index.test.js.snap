// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads DriverProvider 1`] = `<div />`;

exports[`loads Render (with output) 1`] = `
<span>
  Hello
</span>
`;
